gamemodes/terrortown/gamemode/client/cl_targetid.lua

```diff
@@ -172,13 +172,20 @@ function GM:PostDrawTranslucentRenderables(bDrawingDepth, bDrawingSkybox)
 		local ply = plys[i]
 		local rd = ply:GetSubRoleData()
 
-		if ply:IsActive()
+		local shouldDrawDefault = ply:IsActive()
 			and ply:HasRole()
 			and (not client:IsActive() or ply:IsInTeam(client) or rd.isPublicRole)
 			and not rd.avoidTeamIcons
-		then
-			DrawOverheadRoleIcon(ply, rd.iconMaterial, ply:GetRoleColor())
-		end
+
+		---
+		-- @realm client
+		local shouldDraw, material, color = hook.Run("TTT2ModifyOverheadIcon", ply, shouldDrawDefault)
+
+		if shouldDraw == false
+			or not shouldDrawDefault and not (material and color)
```

Comment:
Why the additional and check for material and color? Those do not make sense to me

Reply:
so that it is only used if both are returned. If only one or none are returned, the default OHI is used